x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2239,4 +2260,16 @@ public void testScriptsInsideAggregateFunctions_WithDateField_AndExtendedStats()
             + "InternalSqlScriptUtils.asDateTime(params.a0),InternalSqlScriptUtils.asDateTime(params.v0)))\",\"lang\":\"painless\","
             + "\"params\":{\"v0\":\"2020-05-03T00:00:00.000Z\"}},\"gap_policy\":\"skip\"}}}}}}"));
     }
+
+    public void testInOutOfRangeValues() {
+        QlIllegalArgumentException ex = expectThrows(QlIllegalArgumentException.class,
+            () -> optimizeAndPlan("SELECT int FROM test WHERE int IN (1, 2, 3, " + Long.MAX_VALUE + ", 5, 6, 7)"));
+        assertThat(ex.getMessage(), is("[" + Long.MAX_VALUE + "] out of [integer] range"));
+    }
+
+    public void testInInRangeValues() {
```

Comment:
This might be suitable for `x-pack/plugin/sql/qa/server/src/main/resources/filter.sql-spec`.

Reply:
I can add the positive case to the integration tests too.

Reply:
One problem though: I don't see any `long` type fields in the `test_emp` dataset -> I cannot test it there (and IMHO it does not worth to change the dataset for this one testcase that I can test in a unit test).

Reply:
Not the most straightforward construct but casts should work;  something like `...WHERE salary::LONG IN (1, 2::LONG)`. But anyways, I think the tests are OK as they have been merged too.

Reply:
Good idea, will remember this one next time.